### PR TITLE
[pdu support] unify the PDU support naming convention

### PR DESCRIPTION
**Pull request by yxieca**
### Description of PR
Unify the naming convention of PDU and PSU

Summary:
Fixes # (issue)

### Type of change

- [ ] Bug fix
- [x] Testbed and Framework(new/improvement)
- [ ] Test case(new/improvement)

#### What is the motivation for this PR?
PDU/PSU was fix used in current code. The device external to the DUT that supply power to PSU (on DUT) is PDU. The unit send out power is outlet.

#### How did you do it?
This change unifies this concept in class names, function names and variables.

No functional change is supposed to happen with this change.

Signed-off-by: Ying Xie <ying.xie@microsoft.com>

#### How did you verify/test it?
Run following test and passed:

platform_tests/test_reboot.py::test_power_off_reboot
 platform_tests/test_platform_info.py::test_turn_on_off_psu_and_check_psustatus
snmp/test_snmp_phy_entity.py::test_turn_off_pdu_and_check_psu_info

And validated devutils continue working.

### CODE REVIEW

File: tests/common/plugins/pdu_controller/controller_base.py [hunk 1]
```diff
@@ -1,56 +1,56 @@
 """
-Base class for controlling PSUs of DUT
+Base class for controlling PDUs conecting to DUT
 
-This file defines the base class for controlling PSUs of DUT. The base class defined the basic interface of
-PSU controllers.
+This file defines the base class for controlling PDUs outlets. The base class defined the basic interface of
```
Comment (jleveque) [inline tests/common/plugins/pdu_controller/controller_base.py:4]: s/PDUs/PDU/

File: tests/common/plugins/pdu_controller/controller_base.py [hunk 2]
```diff
@@ -1,56 +1,56 @@
 """
-Base class for controlling PSUs of DUT
+Base class for controlling PDUs conecting to DUT
```
Comment (jleveque) [inline tests/common/plugins/pdu_controller/controller_base.py:2]: Typo. However, suggest rewording: `Base class for controlling PDU(s) connected to DUT power supplies`

File: tests/common/plugins/pdu_controller/controller_base.py [hunk 3]
```diff
@@ -1,56 +1,56 @@
...
 
-This file defines the base class for controlling PSUs of DUT. The base class defined the basic interface of
-PSU controllers.
+This file defines the base class for controlling PDUs outlets. The base class defined the basic interface of
+PDU controllers.
 
-The PSU controller for actually controlling PSUs must be a subclass of the PsuControllerBase class and must
+The PDU controller for actually controlling PDUs must be a subclass of the PduControllerBase class and must
```
Comment (jleveque) [inline tests/common/plugins/pdu_controller/controller_base.py:7]: Suggest removing `for actually controlling PDUs`

File: tests/common/plugins/pdu_controller/controller_base.py [hunk 4]
```diff
@@ -1,56 +1,56 @@
...
+class PduControllerBase():
     """
-    @summary: Base class for PSU controller
+    @summary: Base class for PDU controller
 
-    This base class defines the basic interface to be provided by PSU controller.
+    This base class defines the basic interface to be provided by PDU controller.
 
-    The PSU controller for actually controlling PSUs must be a subclass of the PsuControllerBase class and must
+    The PDU controller for actually controlling PDUs must be a subclass of the PduControllerBase class and must
```
Comment (jleveque) [inline tests/common/plugins/pdu_controller/controller_base.py:21]: Suggest removing `for actually controlling PDUs`

File: tests/common/plugins/pdu_controller/controller_base.py [hunk 5]
```diff
@@ -1,56 +1,56 @@
...
 
-    def get_psu_status(self, psu_id=None):
+    def get_outlet_status(self, outlet=None):
         """
-        @summary: Get current power status of PSUs
+        @summary: Get current power status of PDU outlets
 
-        @param psu_id: Optional PSU ID, it could be integer or string digit. If no psu_id is specified, power status of
-                       all PSUs should be returned
+        @param outlet: Optional PDU ID, it could be integer or string digit. If no outlet is specified, power status of
```
Comment (jleveque) [inline tests/common/plugins/pdu_controller/controller_base.py:49]: s/Optional PDU ID/Optional outlet ID/

File: tests/common/plugins/pdu_controller/controller_base.py [hunk 6]
```diff
@@ -1,56 +1,56 @@
...
+                       all PDUs should be returned
```
Comment (jleveque) [inline tests/common/plugins/pdu_controller/controller_base.py:50]: s/PDUs/outlets/

File: tests/common/plugins/pdu_controller/controller_base.py [hunk 7]
```diff
@@ -1,56 +1,56 @@
...
         @return: Returns a list of dictionaries. For example:
-                     [{"psu_id": 0, "psu_on": True}, {"psu_id": 1, "psu_on": True}]
-                 If getting PSU status failed, an empty list should be returned.
+                     [{"outlet": 0, "pdu_on": True}, {"outlet": 1, "pdu_on": True}]
+                 If getting PDU status failed, an empty list should be returned.
```
Comment (jleveque) [inline tests/common/plugins/pdu_controller/controller_base.py:53]: s/PDU/outlet/

File: tests/common/plugins/pdu_controller/snmp_pdu_controllers.py [hunk 8]
```diff
@@ -168,107 +168,107 @@ def map_host_to_lane(self, lane_id):
...
+        port_oid = self.pPORT_CONTROL_BASE_OID + self.pdu_ports[rfc1902.Integer(outlet)]
         errorIndication, errorStatus, _, _ = \
         cmdgen.CommandGenerator().setCmd(
             cmdgen.CommunityData(self.snmp_rwcommunity),
             cmdgen.UdpTransportTarget((self.controller, 161)),
             (port_oid, rfc1902.Integer(self.CONTROL_ON)),
         )
         if errorIndication or errorStatus != 0:
-            logging.debug("Failed to turn on PSU %s, exception: %s" % (str(psu_id), str(errorStatus)))
+            logging.debug("Failed to turn on PDU %s, exception: %s" % (str(outlet), str(errorStatus)))
```
Comment (jleveque) [inline tests/common/plugins/pdu_controller/snmp_pdu_controllers.py:212]: s/PDU/outlet/

File: tests/common/plugins/pdu_controller/snmp_pdu_controllers.py [hunk 9]
```diff
@@ -168,107 +168,107 @@ def map_host_to_lane(self, lane_id):
...
             return False
         return True
 
-    def turn_off_psu(self, psu_id):
+    def turn_off_outlet(self, outlet):
         """
-        @summary: Use SNMP to turn off power to PSU of DUT specified by psu_id
+        @summary: Use SNMP to turn off power to PDU of DUT specified by outlet
```
Comment (jleveque) [inline tests/common/plugins/pdu_controller/snmp_pdu_controllers.py:218]: s/PDU/PDU outlet/

File: tests/common/plugins/pdu_controller/snmp_pdu_controllers.py [hunk 10]
```diff
@@ -168,107 +168,107 @@ def map_host_to_lane(self, lane_id):
...
 
         DUT hostname must be configured in PDU port name/description. But it is hard to specify which PDU port is
-        connected to the first PSU of DUT and which port is connected to the second PSU.
+        connected to the first PDU of DUT and which port is connected to the second PDU.
```
Comment (jleveque) [inline tests/common/plugins/pdu_controller/snmp_pdu_controllers.py:221]: I think these should remain "PSU"

File: tests/common/plugins/pdu_controller/snmp_pdu_controllers.py [hunk 11]
```diff
@@ -168,107 +168,107 @@ def map_host_to_lane(self, lane_id):
...
 
         Because of this, currently we just find out which PDU ports are connected to PSUs of which DUT. We cannot
         find out the exact mapping between PDU ports and PSUs of DUT.
 
-        To overcome this limitation, the trick is to convert the specified psu_id to integer, then calculate the mode
+        To overcome this limitation, the trick is to convert the specified outlet to integer, then calculate the mode
         upon the number of PSUs on DUT. The calculated mode is used as an index to get PDU ports ID stored in
-        self.pdu_ports. But still, we cannot gurante that psu_id 0 is first PSU of DUT, and so on.
+        self.pdu_ports. But still, we cannot gurante that outlet 0 is first PDU of DUT, and so on.
```
Comment (jleveque) [inline tests/common/plugins/pdu_controller/snmp_pdu_controllers.py:228]: s/PDU/PSU/

Also, existing typo: s/gurante/guarantee/

File: tests/common/plugins/pdu_controller/snmp_pdu_controllers.py [hunk 12]
```diff
@@ -168,107 +168,107 @@ def map_host_to_lane(self, lane_id):
...
 
-        @param psu_id: ID of the PSU on SONiC DUT
+        @param outlet: ID of the PDU on SONiC DUT
```
Comment (jleveque) [inline tests/common/plugins/pdu_controller/snmp_pdu_controllers.py:230]: Here, `psu_id` was meant to be the PSU ID of the DUT, not the outlet ID of the PDU. Hmm...

File: tests/common/plugins/pdu_controller/snmp_pdu_controllers.py [hunk 13]
```diff
@@ -168,107 +168,107 @@ def map_host_to_lane(self, lane_id):
...
+        port_oid = self.pPORT_CONTROL_BASE_OID + self.pdu_ports[rfc1902.Integer(outlet)]
         errorIndication, errorStatus, _, _ = \
         cmdgen.CommandGenerator().setCmd(
             cmdgen.CommunityData(self.snmp_rwcommunity),
             cmdgen.UdpTransportTarget((self.controller, 161)),
             (port_oid, rfc1902.Integer(self.CONTROL_OFF)),
         )
         if errorIndication or errorStatus != 0:
-            logging.debug("Failed to turn on PSU %s, exception: %s" % (str(psu_id), str(errorStatus)))
+            logging.debug("Failed to turn on PDU %s, exception: %s" % (str(outlet), str(errorStatus)))
```
Comment (jleveque) [inline tests/common/plugins/pdu_controller/snmp_pdu_controllers.py:245]: s/PDU/outlet/

File: tests/common/plugins/pdu_controller/snmp_pdu_controllers.py [hunk 14]
```diff
@@ -168,107 +168,107 @@ def map_host_to_lane(self, lane_id):
...
         return True
 
-    def get_psu_status(self, psu_id=None):
+    def get_outlet_status(self, outlet=None):
         """
         @summary: Use SNMP to get status of PDU ports supplying power to PSUs of DUT
 
         DUT hostname must be configured in PDU port name/description. But it is hard to specify which PDU port is
-        connected to the first PSU of DUT and which port is connected to the second PSU.
+        connected to the first PDU of DUT and which port is connected to the second PDU.
```
Comment (jleveque) [inline tests/common/plugins/pdu_controller/snmp_pdu_controllers.py:254]: I think these should remain "PSU"

File: tests/common/plugins/pdu_controller/snmp_pdu_controllers.py [hunk 15]
```diff
@@ -168,107 +168,107 @@ def map_host_to_lane(self, lane_id):
...
 
         Because of this, currently we just find out which PDU ports are connected to PSUs of which DUT. We cannot
         find out the exact mapping between PDU ports and PSUs of DUT.
 
-        To overcome this limitation, the trick is to convert the specified psu_id to integer, then calculate the mode
+        To overcome this limitation, the trick is to convert the specified outlet to integer, then calculate the mode
         upon the number of PSUs on DUT. The calculated mode is used as an index to get PDU ports ID stored in
-        self.pdu_ports. But still, we cannot gurante that psu_id 0 is first PSU of DUT, and so on.
+        self.pdu_ports. But still, we cannot gurante that outlet 0 is first PDU of DUT, and so on.
```
Comment (jleveque) [inline tests/common/plugins/pdu_controller/snmp_pdu_controllers.py:261]: s/PDU/PSU/

File: tests/common/plugins/pdu_controller/snmp_pdu_controllers.py [hunk 16]
```diff
@@ -168,107 +168,107 @@ def map_host_to_lane(self, lane_id):
...
 
-        @param psu_id: Optional. If specified, only return status of PDU port connected to specified PSU of DUT. If
+        @param outlet: Optional. If specified, only return status of PDU port connected to specified PDU of DUT. If
```
Comment (jleveque) [inline tests/common/plugins/pdu_controller/snmp_pdu_controllers.py:263]: s/PDU port/PDU outlet/
s/specified PDU of DUT/specified PSU of DUT/

File: tests/common/plugins/pdu_controller/snmp_pdu_controllers.py [hunk 17]
```diff
@@ -287,21 +287,21 @@ def get_psu_status(self, psu_id=None):
...
-                        status = {"psu_id": idx, "psu_on": True if current_val == self.STATUS_ON else False}
+                        status = {"outlet_id": idx, "outlet_on": True if current_val == self.STATUS_ON else False}
                         results.append(status)
-        if psu_id is not None:
-            idx = int(psu_id) % len(self.pdu_ports)
+        if outlet is not None:
+            idx = int(outlet) % len(self.pdu_ports)
             results = results[idx:idx+1]
-        logging.info("Got PSU status: %s" % str(results))
+        logging.info("Got PDU status: %s" % str(results))
```
Comment (jleveque) [inline tests/common/plugins/pdu_controller/snmp_pdu_controllers.py:295]: s/PDU/outlet/

File: tests/platform_tests/test_platform_info.py [hunk 18]
```diff
@@ -327,12 +327,12 @@ def test_thermal_control_psu_absence(duthosts, rand_one_dut_hostname, psu_contro
     pytest_require(psu_num >= 2, "At least 2 PSUs required for rest of the testing in this case")
 
     logging.info("Create PSU controller for testing")
-    psu_ctrl = psu_controller
+    pdu_ctrl = pdu_controller
 
-    pytest_require(psu_ctrl, "No PSU controller for %s, skip rest of the testing in this case" % duthost.hostname)
+    pytest_require(pdu_ctrl, "No PSU controller for %s, skip rest of the testing in this case" % duthost.hostname)
```
Comment (jleveque) [inline tests/platform_tests/test_platform_info.py:332]: s/PSU/PDU/ ?

File: tests/platform_tests/test_platform_info.py [hunk 19]
```diff
@@ -327,12 +327,12 @@ def test_thermal_control_psu_absence(duthosts, rand_one_dut_hostname, psu_contro
...
 
     logging.info("To avoid DUT being shutdown, need to turn on PSUs that are not powered")
-    turn_all_psu_on(psu_ctrl)
+    turn_all_outlet_on(pdu_ctrl)
```
Comment (jleveque) [inline tests/platform_tests/test_platform_info.py:335]: s/outlet/outlets/

File: tests/platform_tests/test_reboot.py [hunk 20]
```diff
@@ -153,54 +153,54 @@ def _power_off_reboot_helper(kwargs):
     @summary: used to parametrized test cases on power_off_delay
     @param kwargs: the delay time between turning off and on the PSU
     """
-    psu_ctrl = kwargs["psu_ctrl"]
-    all_psu = kwargs["all_psu"]
+    pdu_ctrl = kwargs["pdu_ctrl"]
+    all_outlet = kwargs["all_outlet"]
```
Comment (jleveque) [inline tests/platform_tests/test_reboot.py:157]: s/all_outlet/all_outlets/

File: tests/common/plugins/pdu_controller/controller_base.py [hunk 21]
```diff
@@ -1,56 +1,56 @@
...
+        @summary: Get current power status of PDU outlets
 
-        @param psu_id: Optional PSU ID, it could be integer or string digit. If no psu_id is specified, power status of
-                       all PSUs should be returned
+        @param outlet: Optional PDU ID, it could be integer or string digit. If no outlet is specified, power status of
+                       all PDUs should be returned
         @return: Returns a list of dictionaries. For example:
-                     [{"psu_id": 0, "psu_on": True}, {"psu_id": 1, "psu_on": True}]
-                 If getting PSU status failed, an empty list should be returned.
+                     [{"outlet": 0, "pdu_on": True}, {"outlet": 1, "pdu_on": True}]
```
Comment (jleveque) [inline tests/common/plugins/pdu_controller/controller_base.py:52]: s/pdu_on/outlet_on/ ?

**Review by jleveque — CHANGES_REQUESTED** [review]

File: tests/common/plugins/pdu_controller/controller_base.py (hunk 1 above)
Comment (yxieca) [reply tests/common/plugins/pdu_controller/controller_base.py]: Thanks Joe! Fixed.

File: tests/common/plugins/pdu_controller/controller_base.py (hunk 2 above)
Comment (yxieca) [reply tests/common/plugins/pdu_controller/controller_base.py]: Thanks Joe! Fixed.

File: tests/common/plugins/pdu_controller/controller_base.py (hunk 3 above)
Comment (yxieca) [reply tests/common/plugins/pdu_controller/controller_base.py]: Thanks Joe! Fixed.

File: tests/common/plugins/pdu_controller/controller_base.py (hunk 4 above)
Comment (yxieca) [reply tests/common/plugins/pdu_controller/controller_base.py:21]: Thanks Joe! Fixed.

Comment (yxieca) [reply tests/common/plugins/pdu_controller/controller_base.py]: Actually missed this one in the first round. fixed now.

File: tests/common/plugins/pdu_controller/controller_base.py (hunk 5 above)
Comment (yxieca) [reply tests/common/plugins/pdu_controller/controller_base.py]: Thanks Joe! Fixed.

File: tests/common/plugins/pdu_controller/snmp_pdu_controllers.py (hunk 17 above)
Comment (yxieca) [reply tests/common/plugins/pdu_controller/snmp_pdu_controllers.py]: Thanks Joe! Fixed.

File: tests/platform_tests/test_platform_info.py (hunk 18 above)
Comment (yxieca) [reply tests/platform_tests/test_platform_info.py]: Thanks Joe! Fixed.

File: tests/platform_tests/test_platform_info.py (hunk 19 above)
Comment (yxieca) [reply tests/platform_tests/test_platform_info.py]: Thanks Joe! Fixed.

File: tests/platform_tests/test_reboot.py (hunk 20 above)
Comment (yxieca) [reply tests/platform_tests/test_reboot.py:157]: Thanks Joe. Done in the last commit. Retested reboot power cycle.

File: tests/common/plugins/pdu_controller/controller_base.py (hunk 21 above)
Comment (yxieca) [reply tests/common/plugins/pdu_controller/controller_base.py]: Thanks Joe. Done in the last commit. Retested reboot power cycle.

**Comment by jleveque** [comment]
@yxieca: I assume some of my review comments were hidden to you, because not all were addressed in your latest commit.

**Comment by yxieca** [comment]
> @yxieca: I assume some of my review comments were hidden to you, because not all were addressed in your latest commit.

you are right. I missed on push after running test for all_outlets changes.

**Comment by jleveque** [comment]
> > @yxieca: I assume some of my review comments were hidden to you, because not all were addressed in your latest commit.
> 
> you are right. I missed one push after running test for all_outlets changes.

It looks like there are still unaddressed comments. Are they hidden/compressed for you?

File: tests/common/plugins/pdu_controller/snmp_pdu_controllers.py (hunk 16 above)
Comment (yxieca) [reply tests/common/plugins/pdu_controller/snmp_pdu_controllers.py]: Thanks Joe, addressed.

File: tests/common/plugins/pdu_controller/snmp_pdu_controllers.py (hunk 15 above)
Comment (yxieca) [reply tests/common/plugins/pdu_controller/snmp_pdu_controllers.py]: Thanks Joe, addressed.

File: tests/common/plugins/pdu_controller/snmp_pdu_controllers.py (hunk 14 above)
Comment (yxieca) [reply tests/common/plugins/pdu_controller/snmp_pdu_controllers.py]: Thanks Joe, addressed.

File: tests/common/plugins/pdu_controller/snmp_pdu_controllers.py (hunk 13 above)
Comment (yxieca) [reply tests/common/plugins/pdu_controller/snmp_pdu_controllers.py]: Thanks Joe, addressed.

File: tests/common/plugins/pdu_controller/snmp_pdu_controllers.py (hunk 12 above)
Comment (yxieca) [reply tests/common/plugins/pdu_controller/snmp_pdu_controllers.py]: Changed comments. The comments above stated the fact that hard to build the actual mapping. We will treat this as outlet ID, and figure out mapping in graph.

File: tests/common/plugins/pdu_controller/snmp_pdu_controllers.py (hunk 11 above)
Comment (yxieca) [reply tests/common/plugins/pdu_controller/snmp_pdu_controllers.py]: Thanks Joe, addressed.

File: tests/common/plugins/pdu_controller/snmp_pdu_controllers.py (hunk 10 above)
Comment (yxieca) [reply tests/common/plugins/pdu_controller/snmp_pdu_controllers.py]: Thanks Joe, addressed.

File: tests/common/plugins/pdu_controller/snmp_pdu_controllers.py (hunk 8 above)
Comment (yxieca) [reply tests/common/plugins/pdu_controller/snmp_pdu_controllers.py]: Thanks Joe, addressed.

File: tests/common/plugins/pdu_controller/snmp_pdu_controllers.py (hunk 9 above)
Comment (yxieca) [reply tests/common/plugins/pdu_controller/snmp_pdu_controllers.py]: Thanks Joe, addressed.

File: tests/common/plugins/pdu_controller/controller_base.py (hunk 7 above)
Comment (yxieca) [reply tests/common/plugins/pdu_controller/controller_base.py]: Thanks Joe, addressed.

File: tests/common/plugins/pdu_controller/controller_base.py (hunk 6 above)
Comment (yxieca) [reply tests/common/plugins/pdu_controller/controller_base.py]: Thanks Joe, addressed.

**Review by jleveque — APPROVED** [review]

**Comment by yxieca** [comment]
Addressed the merge conflict
